### PR TITLE
the ability to use the keyboard to open select dropdown, EECORE-1649

### LINKED ISSUE #1686: Accessibility Issues: Control panel publishing > select dropdown menus, radio, and checkbox fields
EE 6.1.6
PHP 7.3.32

I don't normally use a screenreader, and primarily use a mouse for navigation. However, I find it extremely frustrating while publishing content that I can't keyboard-tab into a select dropdown menu. Having to manually click on the dropdown fields is tedious and inaccessible. See screenshot below for the interface that finally "broke the camel's back" and made me submit this ticket.

While getting ready to submit this issue, I tested a couple other fields and found that checkbox fields and toggles are able to be tabbed to via keyboard. However, they don't have focus states, so unless you're using a screenreader, (like the one Mac OS Monterey provides), or an overlay that adds in this UX, you have no idea and are virtually lost until you start tapping enter and see what happens. 

The "Add Category" button **_barely_** has a focus state. 

Also, radio buttons don't appear to be tab-able, (e.g. to change the entry's author). And they don't have a focus state. 

I second the issues submitted (_over a year ago_) by @blue-dreamer here: https://github.com/ExpressionEngine/ExpressionEngine/issues/556
and by chunkitdesign here: https://github.com/ExpressionEngine/ExpressionEngine/issues/1590

Screen Shot 2021-12-31 at 2 08 29 PM

### PR DESCRIPTION
**Pull request by Yulyaswan**
fixes #1686

### CODE REVIEW

File: themes/ee/asset/javascript/src/common.js [hunk 1]
```diff
@@ -9,6 +9,14 @@
 
 $(document).ready(function(){
 
+	$('.js-dropdown-toggle, .select__dropdown-item').each(function() {
+		$(this).keyup(function(event) {
+			if (event.keyCode === 13) {
```
Comment (matthewjohns0n) [inline themes/ee/asset/javascript/src/common.js]: Could you leave a comment labeling keycode 13? It's hard to read in passing without it.

**Review by matthewjohns0n — CHANGES_REQUESTED** [review]

File: themes/ee/asset/javascript/src/common.js (hunk 1 above)
Comment (Yulyaswan) [reply themes/ee/asset/javascript/src/common.js]: Left a comment in the file. keyCode 13 => Enter

File: themes/ee/asset/javascript/src/fields/dropdown/dropdown.js [hunk 2]
```diff
@@ -103,41 +107,42 @@ function (_React$Component) {
 
       var tooMany = this.props.items.length > this.props.tooMany && !this.state.loading;
       var selected = this.state.selected;
-      return React.createElement("div", {
+      return /*#__PURE__*/React.createElement("div", {
```
Comment (intoeetive) [inline themes/ee/asset/javascript/src/fields/dropdown/dropdown.js]: What's the reason for this `/*#__PURE__*/` to be added throughout this file?

File: themes/ee/asset/javascript/src/fields/dropdown/dropdown.js [hunk 3]
```diff
@@ -1,22 +1,26 @@
 "use strict";
 
-function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }
+function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
```
Comment (intoeetive) [inline themes/ee/asset/javascript/src/fields/dropdown/dropdown.js]: This `"@babel/helpers - typeof"` piece sounds strange to me - is this needed here?

Comment (Yulyaswan) [reply themes/ee/asset/javascript/src/fields/dropdown/dropdown.js]: removed this extra code. Rebuilt js files

File: themes/ee/asset/javascript/src/common.js (hunk 1 above)
Comment (matthewjohns0n) [reply themes/ee/asset/javascript/src/common.js]: Thanks!

**Review by matthewjohns0n — DISMISSED** [review]

**Review by intoeetive — CHANGES_REQUESTED** [review]
@Yulyaswan the common convention is to also use spacebar to open the dropdown, which does not work here - can we add it?

**Comment by Yulyaswan** [comment]
Added code for spacebar

**Comment by matthewjohns0n** [comment]
I went and added all the default fieldtypes to a channel to test this. I saw that the select fieldtypes are working with space now, from Yuri's last comment. 

@Yulyaswan I still wasn't able to tab to radio fields, which I think we really should fix with this. 

I also found a few items that were outside the scope of this task that I would like to see us address soon. It details issues with color picker swatches, date picker fieldtype, losing focus properly on the relationship fieldtype, file assets in modals, and specific issues with using grid: https://packettide.atlassian.net/browse/EECORE-1684

**Review by intoeetive — APPROVED** [review]

**Pull request merged**